src/librustc_parse/lexer/mod.rs

```diff
@@ -46,19 +46,10 @@ impl<'a> StringReader<'a> {
         source_file: Lrc<rustc_span::SourceFile>,
         override_span: Option<Span>,
     ) -> Self {
-        // Make sure external source is loaded first, before accessing it.
-        // While this can't show up during normal parsing, `retokenize` may
-        // be called with a source file from an external crate.
-        sess.source_map().ensure_source_file_source_present(Lrc::clone(&source_file));
-
-        let src = if let Some(src) = &source_file.src {
-            Lrc::clone(&src)
-        } else if let Some(src) = source_file.external_src.borrow().get_source() {
```

Comment:
Why was the `external_src` logic removed?

Reply:
I assumed from the comment that it was only needed for `retokenize`. But let me check blame to see if the two were actually added simultaneously.

Reply:
yup, both ensure_source_file_present and this branch were added in https://github.com/rust-lang/rust/pull/70172.

I *think* it should be fine to remove thouse now, as we no longer have retokenize. It might be the case that some other code-path now relies on this path, but this seems unlikely (StringReader is not really used outside of librustc_parse), and, in any case, lexing external files which might or might not be there feels just weird.